lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -935,22 +986,67 @@ def configure_network(self, vm_obj):
                     if 'gateway' in network_devices[key]:
                         guest_map.adapter.gateway = network_devices[key]['gateway']
 
-                    if self.params.get('domain'):
-                        guest_map.adapter.dnsDomain = self.params['domain']
+                # On Windows, DNS domain is set by network interface
+                if 'domain' in network_devices[key]:
```

Comment:
Indentation error.
If there's no `ip:` in the definition, but there is a `domain:` definition, this will fail because `guest_map` and `guest_map.adapter` are not defined.